cache/contenthash/checksum_test.go

```diff
@@ -460,37 +469,43 @@ func TestChecksumIncludeExclude(t *testing.T) {
 	cc, err := newCacheContext(ref.Metadata(), nil)
 	require.NoError(t, err)
 
-	dgst, err := cc.Checksum(context.TODO(), ref, "foo", ChecksumOpts{IncludePatterns: []string{"foo"}}, nil)
-	require.NoError(t, err)
-
-	require.Equal(t, dgstFileData0, dgst)
+	opts := func(opts ChecksumOpts) ChecksumOpts {
+		opts.Wildcard = wildcard
+		return opts
+	}
 
-	dgstFoo, err := cc.Checksum(context.TODO(), ref, "", ChecksumOpts{IncludePatterns: []string{"foo"}}, nil)
+	dgstFoo, err := cc.Checksum(context.TODO(), ref, "", opts(ChecksumOpts{IncludePatterns: []string{"foo"}}), nil)
 	require.NoError(t, err)
-	require.Equal(t, dgstFileData0, dgst)
+	require.NotEqual(t, digest.FromBytes([]byte{}), dgstFoo)
```

Comment:
Not sure why the equality check is gone here.

Reply:
The old `require.Equal` was checking the wrong value (`dgst` instead of `dgstFoo`). It turns out `dgstFileData0` was never being returned in this case. There are two reasons:

- `includedPaths` returns both `/` and `/foo`, so we don't take the hash of `/foo` as the hash to return. I think we could avoid returning `/` in this case - I don't believe we'd ever use the metadata of `/` in a copy (but please correct me if I'm wrong).
- `Checksum` only returns `includedPaths[0].Record.Digest` when `path.Base(p) == path.Base(includedPaths[0].Path)`, but that's not the case here. `p` is an empty string (copy from root) and `includedPaths[0].Path` is `/foo`.

Do you think the change is good based on this explanation, or should we adjust any of this logic?